### PR TITLE
feat(core): confirm ETH stake, unstake, claim

### LINKED ISSUE #3517: Trezor Ethereum staking pool
Dedicated screens for staking, unstaking and claiming on Ethereum.

Bake these contracts+methods into the firmware.

### ABI + Methods
[Pool.json](https://github.com/trezor/trezor-firmware/files/14167831/Pool.json). Methods: stake, unstake
[Accounting.json](https://github.com/trezor/trezor-firmware/files/14171148/Accounting.json). Method: claimWithdrawRequest

### Contracts
```
        case 'mainnet':
            ADDRESS_CONTRACT_ACCOUNTING = '0x7a7f0b3c23C23a31cFcb0c44709be70d4D545c6e';
            ADDRESS_CONTRACT_POOL = '0xD523794C879D9eC028960a231F866758e405bE34';
            break;
        case 'holesky':
            ADDRESS_CONTRACT_ACCOUNTING = '0x624087DD1904ab122A32878Ce9e933C7071F53B9';
            ADDRESS_CONTRACT_POOL = '0xAFA848357154a6a624686b348303EF9a13F63264';
            break;
```

### Source property
The source attribute identifies where the staking is coming from. `1` identifies Trezor users. The dedicated screens should check the contract address and the source attribute for stake and unstake methods.

### Designs
* [Staking](https://www.figma.com/file/hZ22ckyTRlCYRLTNUbTdWO/Exploring-Accounts?type=design&node-id=6417-25787&mode=design&t=kCOc6dRJu9b5xOmA-4)
* [Unstaking](https://www.figma.com/file/hZ22ckyTRlCYRLTNUbTdWO/Exploring-Accounts?type=design&node-id=8237-25857&mode=design&t=kCOc6dRJu9b5xOmA-4)
* [Claiming](https://www.figma.com/file/hZ22ckyTRlCYRLTNUbTdWO/Exploring-Accounts?type=design&node-id=8237-26033&mode=design&t=kCOc6dRJu9b5xOmA-4)

### PR DESCRIPTION
**Pull request by obrusvit**
This PR adds specific screens for confirming `stake`, `unstake` and `claim` contracts for Everstake. 
Resolves #3517

### CODE REVIEW

File: core/src/apps/ethereum/sign_tx.py
Comment (grdddj) [inline core/src/apps/ethereum/sign_tx.py:8]: I would consider importing it `as constants`, so all the code reads `constants.XYZ`

File: core/src/apps/ethereum/layout.py
Comment (grdddj) [inline core/src/apps/ethereum/layout.py:60]: I think we could remove majority (maybe all) of the kwargs to save some storage bytes.

It is true that there is a lot of args, so kwargs have documentation function, but we generally do not use them where not necessary.

(Description can be achieved also with args, e.g. like this:
```python
await confirm_ethereum_staking_tx(
    TR.ethereum__staking_unstake,  # title
    TR.ethereum__staking_unstake_intro,  # intro_question
    TR.ethereum__staking_unstake,  # verb
    ...
)
```
)

File: core/src/apps/ethereum/layout.py
Comment (grdddj) [inline core/src/apps/ethereum/layout.py:261]: I am surprised that moving this function did not break unittests - in `core/tests/test_apps.ethereum.layout.py`  we are importing `format_ethereum_amount` still from this file.

(As it is imported here also, maybe it works, but we should change the import to `from helpers`)

File: core/src/apps/ethereum/helpers.py
Comment (grdddj) [inline core/src/apps/ethereum/helpers.py:141]: I guess no need to initialize the variable, when we can just return its content right away

**Review by grdddj — COMMENTED** [review]
Some mostly code-related comments.

Also, BTC-only check is failing - https://gitlab.com/satoshilabs/trezor/trezor-firmware/-/jobs/6185787496 - need to whitelist this function in the checking script

File: core/src/apps/ethereum/sign_tx.py
Comment (obrusvit) [reply core/src/apps/ethereum/sign_tx.py]: https://github.com/trezor/trezor-firmware/pull/3534/commits/b3eb30b259e2034a10e613d2f20c7761538a91f7

File: core/src/apps/ethereum/sign_tx.py
Comment (obrusvit) [reply core/src/apps/ethereum/sign_tx.py]: https://github.com/trezor/trezor-firmware/pull/3534/commits/b3eb30b259e2034a10e613d2f20c7761538a91f7

File: core/src/apps/ethereum/sign_tx.py
Comment (obrusvit) [reply core/src/apps/ethereum/sign_tx.py]: https://github.com/trezor/trezor-firmware/pull/3534/commits/b3eb30b259e2034a10e613d2f20c7761538a91f7

File: core/src/apps/ethereum/sign_tx.py
Comment (obrusvit) [reply core/src/apps/ethereum/sign_tx.py]: https://github.com/trezor/trezor-firmware/pull/3534/commits/b3eb30b259e2034a10e613d2f20c7761538a91f7

File: core/src/apps/ethereum/layout.py
Comment (obrusvit) [reply core/src/apps/ethereum/layout.py:60]: Ok, thanks, I kept the names of all the arguments in the comments for consistency.
https://github.com/trezor/trezor-firmware/pull/3534/commits/b3eb30b259e2034a10e613d2f20c7761538a91f7

File: core/src/apps/ethereum/layout.py
Comment (obrusvit) [reply core/src/apps/ethereum/layout.py:261]: Yep, it picked up the import from `layout.py`. I fixed it so that it's imported directly.
https://github.com/trezor/trezor-firmware/pull/3534/commits/b3eb30b259e2034a10e613d2f20c7761538a91f7

File: core/src/apps/ethereum/helpers.py
Comment (obrusvit) [reply core/src/apps/ethereum/helpers.py]: https://github.com/trezor/trezor-firmware/pull/3534/commits/b3eb30b259e2034a10e613d2f20c7761538a91f7

**Comment by obrusvit** [comment]
From review of @Hannsek 

Added title to "Claim" summary:
image

Info from all operations does not have right "Back" but only left "<":
info_back

**Comment by obrusvit** [comment]
rebased on `main` with `autosquash`

File: core/src/trezor/ui/layouts/tr/__init__.py
Comment (grdddj) [inline core/src/trezor/ui/layouts/tr/__init__.py:1070]: Looks like `confirm_ethereum_staking_tx` in `TR` and `TT` code have now different signature --- the position of `chunkify` differs. Also, why is the value of `chunkify` not used in `TR`'s case?

File: core/src/trezor/ui/layouts/tt/__init__.py
Comment (grdddj) [inline core/src/trezor/ui/layouts/tt/__init__.py:1061]: Are there some tests of clicking the info button and seeing the details? Quickly looked at some test in CI and did not see any - https://satoshilabs.gitlab.io/-/trezor/trezor-firmware/-/jobs/6216595229/artifacts/test_ui_report/index.html

Comment (obrusvit) [reply core/src/trezor/ui/layouts/tt/__init__.py:1061]: Yes, good catch in these two comments. 
I will use chunkify for Staking/Claiming address screens on TS3 and add Click tests which verify it.

File: core/src/trezor/ui/layouts/tr/__init__.py
Comment (obrusvit) [reply core/src/trezor/ui/layouts/tr/__init__.py:1072]: `chunkify` is used on TS3. I also found that `confirm_value` can be used in TT with just additional params.
https://github.com/trezor/trezor-firmware/pull/3534/commits/4b4d1fadd319297eb89c383b45fc509650624c0e

File: core/src/trezor/ui/layouts/tt/__init__.py
Comment (obrusvit) [reply core/src/trezor/ui/layouts/tt/__init__.py]: Clicking the info button added
https://github.com/trezor/trezor-firmware/pull/3534/commits/6f395b9859bf10da578a889310ca4548c03a5d95

**Review by grdddj — APPROVED** [review]
LGTM code-wise!

**Comment by obrusvit** [comment]
Force pushed because I forgot to update fixtures after fix of `confirm_value` on TS3
Only the verb changed in confirm button:
image

**Comment by obrusvit** [comment]
Fixtures for all languages updated. 
Should be ready to merge if CI green.

File: core/src/apps/ethereum/sign_tx.py
Comment (matejcik) [inline core/src/apps/ethereum/sign_tx.py:351]: clearer way to spell those:

```
try:
   a = r.read_memoryview(...)
   b = r.read_memoryview(...)
   # ...
   if r.remaining_count():
      raise ValueError
except (ValueError, EofError):
   raise DataError(...)
```

please change that here, you can also change it in the other instance but it's not that much better when there is just a single call to the reader.

File: core/src/apps/ethereum/sign_tx.py
Comment (matejcik) [inline core/src/apps/ethereum/sign_tx.py:375]: always provide text for the DataError. (you can use the same one across all functions, something like "Invalid smart contract call" seems appropriate)
DataError strings are not supposed to be translated (for now)

File: core/src/apps/ethereum/sign_tx.py
Comment (matejcik) [inline core/src/apps/ethereum/sign_tx.py:182]: we prefer early returns to chaining if/elifs, as it makes it easier to be sure that some case doesn't fall through
```
if some_cond:
   if sig == xyz:
       await do_handle_xyz()
       return True
  if sig == abc:
       await do_handle_abc()
       return true

if some_other_cond:
    ...

# single final top-level catch-all
# no early-return happened so this was definitely not handled
return False
```

File: core/src/apps/ethereum/sign_tx.py
Comment (matejcik) [inline core/src/apps/ethereum/sign_tx.py:152]: if the user sends `to` value as non-checksummed address, or omits the 0x prefix, this will fail.

we should probably first decode the address to bytes, use the result here, and specify the pool addresses as bytes too

File: core/src/trezor/wire/errors.py
Comment (matejcik) [inline core/src/trezor/wire/errors.py:22]: see above, there should be no "generic data errors" without a reason

File: core/translations/fr.json
Comment (matejcik) [inline core/translations/fr.json:325]: should we autotranslate these new strings? or should we leave them for the translators?
cc @Hannsek

File: tests/input_flows_helpers.py
Comment (matejcik) [inline tests/input_flows_helpers.py:445]: aside (not blocking the PR on it): this is absolutely horrible. I have zero idea what's going on here and if the test failed on me I wouldn't know how and why.

we MUST figure out a way to spell these test cases in some way that a human person can read

File: core/src/trezor/ui/layouts/tt/__init__.py
Comment (matejcik) [inline core/src/trezor/ui/layouts/tt/__init__.py:862]: style

File: core/embed/rust/src/ui/model_tt/layout.rs
Comment (matejcik) [inline core/embed/rust/src/ui/model_tt/layout.rs:466]: `ConfirmBlob` should by definition be monospace.

why are you using `ConfirmBlob` if you want non-mono text?

**Review by matejcik — CHANGES_REQUESTED** [review]

File: core/embed/rust/src/ui/model_tt/layout.rs
Comment (obrusvit) [reply core/embed/rust/src/ui/model_tt/layout.rs:466]: `ConfirmBlobParams` is used also for `confirm_value`, which is used there. 
I had a problem with `confirm_text`, I'll re-check what was it.

File: core/src/apps/ethereum/sign_tx.py
Comment (obrusvit) [reply core/src/apps/ethereum/sign_tx.py]: Ok, provided message for `DataError` https://github.com/trezor/trezor-firmware/pull/3534/commits/339da2725da19a08e116a30fa035fcb17f31a463

File: core/src/apps/ethereum/sign_tx.py
Comment (obrusvit) [reply core/src/apps/ethereum/sign_tx.py]: Changed https://github.com/trezor/trezor-firmware/pull/3534/commits/339da2725da19a08e116a30fa035fcb17f31a463

File: core/src/apps/ethereum/sign_tx.py
Comment (obrusvit) [reply core/src/apps/ethereum/sign_tx.py]: Good catch, thanks. https://github.com/trezor/trezor-firmware/pull/3534/commits/339da2725da19a08e116a30fa035fcb17f31a463

File: core/src/trezor/wire/errors.py
Comment (obrusvit) [reply core/src/trezor/wire/errors.py]: https://github.com/trezor/trezor-firmware/pull/3534/commits/339da2725da19a08e116a30fa035fcb17f31a463

File: core/embed/rust/src/ui/model_tt/layout.rs
Comment (obrusvit) [reply core/embed/rust/src/ui/model_tt/layout.rs:466]: As discussed on Slack. 
Other options do not allow using :information_source:  button with Stake/Claim address. 
We will keep it for now but a better solution might to provide info items for `confirm_action`

File: tests/input_flows_helpers.py
Comment (obrusvit) [reply tests/input_flows_helpers.py:445]: I kinda agree and I was also getting used to it based on examples from other operations. I guess, one needs to know what the screens are supposed to look like.
The point here is to enter screens not necessarily seen by a user (info items) to validate that the staking address is there and that it's correctly `chunkified` (captured in `fixtures.json`)

Willing to discuss more readable ways how to do it.

File: core/src/apps/ethereum/sign_tx.py
Comment (matejcik) [inline core/src/apps/ethereum/sign_tx.py:111]: very nit but please rename `to_address` and `address_to` in all cases to `address_bytes` so that this is consistent across the whole file.
(`address_bytes` is not _the_ best name but it was here before)

Comment (obrusvit) [reply core/src/apps/ethereum/sign_tx.py]: fixed in rebase

**Comment by obrusvit** [comment]
All green except `legacy upgrade test`: https://gitlab.com/satoshilabs/trezor/trezor-firmware/-/jobs/6245451881 
Jiri mentioned it here: https://satoshilabs.slack.com/archives/C054XGU77QV/p1707138449213669 , so ignoring for now.

**Comment by bosomt** [comment]
**QA OK**

tested stake and unstake, claim is currently unavailable 

**Info:**
 - Suite version: web 24.3.1 (e1bbd6e83b8d33b64dd562a137b279c66c902350)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:123.0) Gecko/20100101 Firefox/123.0
 - OS: MacIntel
 - Screen: 1512x982
 - Device: Trezor T2B1 2.7.0 regular (revision 45e8a842a31e62a6d43d7f6ccac62a45e1198ef0)
 - Transport: BridgeTransport 2.0.33

**Pull request merged**
